fabtests/include/shared.h

```diff
@@ -526,11 +526,11 @@ ssize_t ft_rx(struct fid_ep *ep, size_t size);
 ssize_t ft_tx(struct fid_ep *ep, fi_addr_t fi_addr, size_t size, void *ctx);
 ssize_t ft_inject(struct fid_ep *ep, fi_addr_t fi_addr, size_t size);
 ssize_t ft_post_rma(enum ft_rma_opcodes op, struct fid_ep *ep, size_t size,
-		struct fi_rma_iov *remote, void *context);
+		struct fi_rma_iov *remote, void *context, size_t offset);
```

Comment:
It would make more sense to just pass in the buffer directly, rather than using a global and giving an offset.

Reply:
hm, but the ft_post_rma function applies the same offset to both the local and remote addr, so I'd need to calculate that offset from the global buffer anyways.  After all these patches, post_rma always uses tx_buf as the pointer, so I guess that's not too bad, but as of this commit the post_rma function itself decided if it was going to use tx_buf or rx_buf, and so offset seemed more natural.

You would prefer I calculate remote addr based on the pointer-difference from tx_buf?

Reply:
I'd like to start updating these internal calls to be more maintainable and obvious in how they are used. And since you're needing to change this call, it's a good time to update it.  A function signature of:
```
ft_post_rma(op, ep, buf, size, remote, context);
```
is more intuitive to use.  An offset parameter that applies to both the remote parameter and a global source buffer which may be the tx or may be the rx is confusing.

It's questionable if we should even keep this generic call, versus having separate calls, such as ft_post_write, ft_post_writedata, ft_post_read, and ft_post_write_inject.

Reply:
@shefty Please let me know what you think of the refactoring of ft_post_rma's arguments.  It keeps the arguments to post_rma simple, but it makes some assumptions, namely that every rma operation will occur within tx_buf/rx_buf, and the remote side should be in the opposite buffer (rx->tx, tx->rx).

These assumptions are guarded by an assert, and have been true previously, because the function chose tx_buf or rx_buf based on the RMA operation.